cpp/include/Ice/Connection.h

```diff
@@ -227,7 +227,7 @@ namespace Ice
     class ICE_API Connection
     {
     public:
-        virtual ~Connection() = default;
+        virtual ~Connection() = 0;
```

Comment:
Since it's ICE_API _and_ has out-of-line definitions, may as well put the destructor out of line too.

And yes, = 0 for a destructor means it's pure and out of line. C++ is weird.